### PR TITLE
DAC6-2734: What is the email address for [FI first contact name]

### PR DESCRIPTION
**Pull request by C-Thomp**
[DAC6-2734](https://jira.tools.tax.service.gov.uk/browse/DAC6-2734)
error messages check
image
image

Pulls in Contact Name ->
image

### CODE REVIEW

File: app/controllers/FirstContactEmailController.scala
Comment (C-Thomp) [inline app/controllers/FirstContactEmailController.scala:1]: keep seeing this on most files while optimising imports

**Comment by C-Thomp** [comment]
dynamic name parameter requires name page (DAC6-2733) to exist

File: README.md
Comment (C-Thomp) [inline README.md:1]: to match prod.routes

File: app/views/FirstContactEmailView.scala.html [hunk 1]
```diff
@@ -24,9 +24,11 @@
     govukButton: GovukButton
 )
 
-@(form: Form[_], mode: Mode)(implicit request: Request[_], messages: Messages)
+@(form: Form[_], mode: Mode, name: String = "[SOME BLOKE]")(implicit request: Request[_], messages: Messages)
```
Comment (C-Thomp) [inline app/views/FirstContactEmailView.scala.html:27]: the name page is not in the skeleton so DAC6-2733 must be complete before adding this dynamic text

**Review by matthewharris1 — CHANGES_REQUESTED** [review]

File: app/forms/FirstContactEmailFormProvider.scala [hunk 2]
```diff
@@ -16,17 +16,19 @@
...
+import javax.inject.Inject
 
 class FirstContactEmailFormProvider @Inject() extends Mappings {
 
   def apply(): Form[String] =
     Form(
       "value" -> text("firstContactEmail.error.required")
-        .verifying(maxLength(100, "firstContactEmail.error.length"))
+        .verifying(maxLength(132, "firstContactEmail.error.length"))
+        .verifying(Constraints.emailAddress(errorMessage = "firstContactEmail.error.format"))
```
[the remark this answers is not in the document]
Comment (C-Thomp) [reply app/forms/FirstContactEmailFormProvider.scala:31]: ah thats right, that validatedText field mapping didnt exist when i did this -
Constraints are a part of the Play Framework so get all this stuff for free really, you can apply them to a field with the .verifying method 👍

**Review by matthewharris1 — APPROVED** [review]

**Pull request merged**